plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -1070,21 +1071,26 @@ private static Optional<DataCompression> getTableDataCompression(Handle handle,
 
     private static Optional<DataCompression> getTableDataCompressionWithRetries(Handle handle, JdbcTableHandle table)
     {
-        // DDL operations can take out locks against system tables causing the `getTableDataCompression` query to deadlock
-        final int maxAttemptCount = 3;
-        RetryPolicy<Optional<DataCompression>> retryPolicy = new RetryPolicy<Optional<DataCompression>>()
+        return retryOnDeadlock(() -> getTableDataCompression(handle, table), "error when getting table compression info for '%s'".formatted(table));
+    }
+
+    public static <T> T retryOnDeadlock(CheckedSupplier<T> supplier, String exceptionMessageSuffix)
+    {
+        // DDL operations can take out locks against system tables causing queries against them to deadlock
```

Comment:
javadoc for the method?

Reply:
The javadoc I came up with looked very redundant with the method name - did not add.

```
Retry the supplier up to 3 times if the failure is due to a deadlock on SQL Server.
```

Reply:
I meant only to move that comment to javadoc. Nothing else :)